src/Compilers/Core/Portable/CommandLine/CommonCommandLineParser.cs

```diff
@@ -485,30 +485,36 @@ internal static string RemoveTrailingSpacesAndDots(string path)
                 parsedArgs = newArgs;
                 return true;
             }
-        }
-
-        internal static bool IsClientArgsOption(string arg, string optionName, out bool hasValue, out string optionValue)
-        {
-            hasValue = false;
-            optionValue = null;
 
-            if (!arg.StartsWith(optionName, StringComparison.OrdinalIgnoreCase))
+            bool isClientArgsOption(string arg, string optionName, out bool hasValue, out string optionValue)
             {
-                return false;
-            }
+                hasValue = false;
+                optionValue = null;
 
-            if (arg.Length > optionName.Length && !(arg[optionName.Length] == ':' || arg[optionName.Length] == '='))
-            {
-                return false;
-            }
+                if (arg.Length == 0 || !(arg[0] == '/' || arg[0] == '-'))
+                {
+                    return false;
+                } 
+                
+                arg = arg.Substring(1);
```

Comment:
I'll probably get an answer of "I wish", but can we use Span here to avoid creating copies of the string? Or a different comparison method?

Reply:
Span requires a new reference, which is out scope of this PR, but yes I'd like to add a reference to System.Memory. 😄

Reply:
Perhaps
```
arg.IndexOf(optionName,
   1, optionName.Length - 1,
    StringComparison.OrdinalIgnoreCase) != 1
```

---
In reply to: [229858522](https://github.com/dotnet/roslyn/pull/30818#discussion_r229858522) [](ancestors = 229858522)